### PR TITLE
Enables block hash use for getTransactionCount

### LINKED ISSUE #1856: Add `blockParam` to `eth_getTransactionCount`
### Problem

Add `blockParam` to `eth_getTransactionCount` according to [EIP-1898](https://eips.ethereum.org/EIPS/eip-1898).

### Solution

Extend parameters, validations, tests and all related.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by konstantinabl**
**Description**:

This PR modifies:

* Changes validation for getTransactionCount so we support blockHash
* Changes implementation in eth.ts in order to support the block hash and handle incoming requests
* Adds tests for the new logic

**Related issue(s)**:

Fixes #1856 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by ebadiere — APPROVED** [review]
LG

**Review by AlfredoG87 — APPROVED** [review]

**Review by natanasow — DISMISSED** [review]
LGTM, one question/suggestion :thinking:

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -2208,9 +2210,22 @@ export class EthImpl implements Eth {
...
-    blockNum: number,
+    blockNumOrHash: number | string,
     requestIdPrefix?: string,
   ): Promise<string> {
+    let getBlock;
+
+    try {
+      getBlock = await this.mirrorNodeClient.getBlock(blockNumOrHash);
+    } catch (e) {
+      throw e;
```
[the remark this answers is not in the document]
Comment (konstantinabl) [reply packages/relay/src/lib/eth.ts:2221]: @natanasow  I am wondering isnt there a case where we would get null from the request and that is why we have the === null check?

**Review by natanasow — DISMISSED** [review]
LGTM

File: packages/relay/src/lib/eth.ts [hunk 2]
```diff
@@ -2208,9 +2210,16 @@ export class EthImpl implements Eth {
 
   private async getAccountNonceForHistoricBlock(
     address: string,
-    blockNum: number,
+    blockNumOrHash: number | string,
     requestIdPrefix?: string,
   ): Promise<string> {
+    const getBlock = await this.mirrorNodeClient.getBlock(blockNumOrHash);
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:2216]: Q: do we need to get the block each time?
Please make sure in the flow we're not adding a new mirror node call each time that's no always needed

**Review by Nana-EC — COMMENTED** [review]
Looking good, added one consideration to confirm

**Review by AlfredoG87 — DISMISSED** [review]
LGTM

File: packages/relay/src/lib/eth.ts [hunk 3]
```diff
@@ -1304,8 +1304,10 @@ export class EthImpl implements Eth {
         nonceCount = await this.getAccountLatestEthereumNonce(address, requestIdPrefix);
       } else if (blockNumOrTag === EthImpl.blockEarliest) {
         nonceCount = await this.getAccountNonceForEarliestBlock(requestIdPrefix);
-      } else if (!isNaN(blockNum)) {
+      } else if (!isNaN(blockNum) && blockNumOrTag.length != 66 && blockNum > 0) {
         nonceCount = await this.getAccountNonceForHistoricBlock(address, blockNum, requestIdPrefix);
+      } else if (blockNumOrTag.length == 66 && blockNumOrTag.startsWith('0x')) {
```
Comment (ebadiere) [inline packages/relay/src/lib/eth.ts:1309]: Nit: Is there a reason 66 is not a descriptive numeric string literal? HASH_LENGTH

**Review by ebadiere — APPROVED** [review]
One nit.  LG.

File: packages/relay/src/lib/eth.ts (hunk 3 above)
Comment (AlfredoG87) [reply packages/relay/src/lib/eth.ts:1309]: we could also make a constant out of `0x`

Comment (AlfredoG87) [reply packages/relay/src/lib/eth.ts]: I've made the change 👍 
@ebadiere

**Review by ebadiere — APPROVED** [review]
LG.

**Review by AlfredoG87 — APPROVED** [review]

**Review by AlfredoG87 — APPROVED** [review]

**Pull request merged**